### PR TITLE
Add EarlGreyV2 example project for Pods.

### PR DESCRIPTION
**Pull request by brettfazio**

### CODE REVIEW

File: Demo/EarlGreyExample/EarlGreyExampleUITests/EarlGreyExampleUITests.m [hunk 1]
```diff
@@ -0,0 +1,212 @@
+//
```
Comment (tirodkar) [inline Demo/EarlGreyExample/EarlGreyExampleUITests/EarlGreyExampleUITests.m]: Change this to our License blurb from the other Swift files.

File: Demo/EarlGreyExample/EarlGreyExampleUITests/EarlGreyExampleUITests.m [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+@end
+
+@interface EGUITests : XCTestCase
+- (id<GREYMatcher>) matcherForThursdays;
+@end
+
+@implementation EGUITests
+
+- (void)setUp {
+  // Put setup code here. This method is called before the invocation of each test method in the class.
```
Comment (tirodkar) [inline Demo/EarlGreyExample/EarlGreyExampleUITests/EarlGreyExampleUITests.m]: Nix.

File: Demo/EarlGreyExample/EarlGreyExampleUITests/EarlGreyExampleUITests.m [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+  // In UI tests it is usually best to stop immediately when a failure occurs.
+  self.continueAfterFailure = NO;
+
+  // UI tests must launch the application that they test. Doing this in setup will make sure it happens for each test method.
+  [[[XCUIApplication alloc] init] launch];
+
+  // In UI tests it’s important to set the initial state - such as interface orientation - required for your tests before they run. The setUp method is a good place to do this.
+}
+
+- (void)tearDown {
```
Comment (tirodkar) [inline Demo/EarlGreyExample/EarlGreyExampleUITests/EarlGreyExampleUITests.m]: nix

File: Demo/EarlGreyExample/EarlGreyExampleUITests/EarlGreyExampleUITests.m [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+}
+
+- (void)testBasicSelection {
+  // Select the button with Accessibility ID "clickMe".
+  [EarlGrey selectElementWithMatcher:grey_accessibilityID(@"ClickMe")];
+}
+
+- (void)testBasicSelectionAndAction {
+  // Select and tap the button with Accessibility ID "clickMe".
+  [[EarlGrey selectElementWithMatcher:grey_accessibilityID(@"ClickMe")]
```
Comment (tirodkar) [inline Demo/EarlGreyExample/EarlGreyExampleUITests/EarlGreyExampleUITests.m]: Format:
```
[[EarlGrey selectElementWithMatcher:grey_accessibilityID(@"ClickMe")]
  performAction:grey_tap()];
```

File: Demo/EarlGreyExample/EarlGreyExampleUITests/EarlGreyExampleUITests.m [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+  // Select the button with Accessibility ID "clickMe" and assert it's visible.
+  [[EarlGrey selectElementWithMatcher:grey_accessibilityID(@"ClickMe")]
+   assertWithMatcher:grey_sufficientlyVisible()];
+}
+
+- (void)testBasicSelectionActionAssert {
+  // Select and tap the button with Accessibility ID "clickMe", then assert it's visible.
+  [[[EarlGrey selectElementWithMatcher:grey_accessibilityID(@"ClickMe")]
+    performAction:grey_tap()]
+   assertWithMatcher:grey_sufficientlyVisible()];
```
Comment (tirodkar) [inline Demo/EarlGreyExample/EarlGreyExampleUITests/EarlGreyExampleUITests.m]: Format as stated above.

File: Demo/EarlGreyExample/EarlGreyExampleUITests/EarlGreyExampleUITests.m [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+                     element.alpha = 0.0;}
+                   completion: ^(BOOL finished) {
+                     [UIView animateWithDuration:1.0
+                                           delay:0.0
+                                         options:UIViewAnimationOptionCurveEaseIn
+                                      animations: ^{
+                                        element.alpha = 1.0;}
+                                      completion: nil];
+                   }];
+
```
Comment (tirodkar) [inline Demo/EarlGreyExample/EarlGreyExampleUITests/EarlGreyExampleUITests.m]: nix.

File: Demo/EarlGreyExample/EarlGreyExampleUITests/EarlGreyExampleUITests.m [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+                           *errorOrNil =
+                           [NSError errorWithDomain:kGREYInteractionErrorDomain
+                                               code:kGREYInteractionElementNotFoundErrorCode
+                                           userInfo:nil];
+                           return NO;
+                         }
+                         return element.alpha == expectedAlpha;
+                       }];
+}
+
```
Comment (tirodkar) [inline Demo/EarlGreyExample/EarlGreyExampleUITests/EarlGreyExampleUITests.m]: nix

File: Demo/EarlGreyExample/EarlGreyExampleUITests/EarlGreyExampleUITests.m [hunk 8]
```diff
@@ -0,0 +1,212 @@
...
+  BOOL success = [myCondition waitWithTimeout:5];
+  if (!success) {
+    // Just printing for the example.
+    NSLog(@"Condition not met");
+  } else {
+    [[EarlGrey selectElementWithMatcher:grey_accessibilityID(@"ClickMe")]
+     performAction:grey_tap()];
+  }
+}
+
```
Comment (tirodkar) [inline Demo/EarlGreyExample/EarlGreyExampleUITests/EarlGreyExampleUITests.m]: nix

File: Demo/EarlGreyExample/EarlGreyExampleUITests/EarlGreyExampleUITests.m [hunk 9]
```diff
@@ -0,0 +1,213 @@
...
+@implementation EGUITests
+
+- (void)setUp {
+  self.continueAfterFailure = NO;
+
+  // UI tests must launch the application that they test. Doing this in setup will make sure it happens for each test method.
+  [[[XCUIApplication alloc] init] launch];
+}
+
+- (void)tearDown {
```
Comment (tirodkar) [inline Demo/EarlGreyExample/EarlGreyExampleUITests/EarlGreyExampleUITests.m]: Delete the whole method.

File: Demo/EarlGreyExample/EarlGreyExampleUITests/EarlGreyExampleUITests.m [hunk 10]
```diff
@@ -0,0 +1,213 @@
...
+  [[[EarlGrey selectElementWithMatcher:grey_accessibilityID(@"Cell30")]
+    usingSearchAction:grey_scrollInDirection(kGREYDirectionDown, 50)
+    onElementWithMatcher:grey_accessibilityID(@"table")]
+   performAction:grey_doubleTap()];
+
+  // Move back to top of the table.
+  [[[EarlGrey selectElementWithMatcher:grey_accessibilityID(@"Cell1")]
+    usingSearchAction:grey_scrollInDirection(kGREYDirectionUp, 500)
+    onElementWithMatcher:grey_accessibilityID(@"table")]
+   performAction:grey_doubleTap()];
```
Comment (tirodkar) [inline Demo/EarlGreyExample/EarlGreyExampleUITests/EarlGreyExampleUITests.m]: format

File: Demo/EarlGreyExample/EarlGreyExampleSwift/ViewController.swift [hunk 11]
```diff
@@ -0,0 +1,148 @@
+//
+// Copyright 2016 Google Inc.
```
Comment (tirodkar) [inline Demo/EarlGreyExample/EarlGreyExampleSwift/ViewController.swift]: 2019

File: Demo/EarlGreyExample/EarlGreyExampleSwift/AppDelegate.swift [hunk 12]
```diff
@@ -0,0 +1,28 @@
+//
+// Copyright 2016 Google Inc.
```
Comment (tirodkar) [inline Demo/EarlGreyExample/EarlGreyExampleSwift/AppDelegate.swift]: 2019

File: Demo/EarlGreyExample/EarlGreyExample/Info.plist [hunk 13]
```diff
@@ -0,0 +1,69 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+  <key>license</key>
+  <string>
+
+  Copyright 2016 Google Inc.
```
Comment (tirodkar) [inline Demo/EarlGreyExample/EarlGreyExample/Info.plist]: 2019

File: Demo/EarlGreyExample/EarlGreyExampleUITests/EarlGreyExampleUITests.m (hunk 10 above)
Comment (brettfazio) [reply Demo/EarlGreyExample/EarlGreyExampleUITests/EarlGreyExampleUITests.m]: Done.

File: Demo/EarlGreyExample/EarlGreyExampleSwift/ViewController.swift (hunk 11 above)
Comment (brettfazio) [reply Demo/EarlGreyExample/EarlGreyExampleSwift/ViewController.swift]: Done.

File: Demo/EarlGreyExample/EarlGreyExampleUITests/EarlGreyExampleUITests.m (hunk 9 above)
Comment (brettfazio) [reply Demo/EarlGreyExample/EarlGreyExampleUITests/EarlGreyExampleUITests.m]: Deleted.

File: Demo/EarlGreyExample/EarlGreyExampleSwift/AppDelegate.swift (hunk 12 above)
Comment (brettfazio) [reply Demo/EarlGreyExample/EarlGreyExampleSwift/AppDelegate.swift]: Done.

File: Demo/EarlGreyExample/EarlGreyExample/Info.plist (hunk 13 above)
Comment (brettfazio) [reply Demo/EarlGreyExample/EarlGreyExample/Info.plist]: Done.

File: Demo/EarlGreyExample/EarlGreyExampleUITests/EarlGreyExampleUITests.m [hunk 14]
```diff
@@ -0,0 +1,209 @@
...
+    performAction:grey_doubleTap()];
+}
+
+- (void)testWithInRoot {
+  // Second way to disambiguate: use inRoot to focus on a specific window or container.
+  // There are two buttons with accessibility id "Send", but only one is inside SendMessageView.
+
+  [[[EarlGrey selectElementWithMatcher:grey_accessibilityID(@"Send")]
+    inRoot:grey_kindOfClassName(@"EarlGreyExampleSwift.SendMessageView")]
+   performAction:grey_doubleTap()];
```
Comment (tirodkar) [inline Demo/EarlGreyExample/EarlGreyExampleUITests/EarlGreyExampleUITests.m]: Fix.

**Pull request merged**